### DISCUSSION TITLE
Llava model with images through API

### DISCUSSION THREAD

**Discussion by syaoran68** [oobabooga/text-generation-webui#4626] [General]
Hello I'm currently trying to get the API to ingest images and reply back with responses. 

Model: The bloke llava v1.5 13b GPT
using the new openAI API 

not sure what the API format should be for allowing text-generation-webui to ingest images through the API? 

I've used the openAI vision JSON format and it doesn't recognize that..

Also have used the older API with /api/v1/generate which does generate a response but its quite inconsistent. maybe only getting a positive response every 1/5 times. 

is the openAI API endpoint supporting the openAI vision formatting yet? i noticed it was not on the compatibility list for the openAI API section on the WIKI area. 

Thanks! 

Syaoran.

**Comment by Katehuuh** [comment]
<details open><summary>I modify <a href="https://github.com/oobabooga/text-generation-webui/tree/main/extensions/multimodal#usage-through-api">original multimodal api example</a> with <a href="https://github.com/oobabooga/text-generation-webui/wiki/12-‐-OpenAI-API#python-chat-example">new OpenAI-API example</a> and <a href="https://github.com/oobabooga/text-generation-webui/blob/main/instruction-templates/LLaVA-v1.yaml">replace template LLaVA with v1</a></summary><br>

```python
# --model TheBloke_llava-v1.5-13B-GPTQ --multimodal-pipeline llava-v1.5-13b --extensions multimodal --loader autogptq
import base64
import requests

def process_image(image_path: str) -> str:
    CONTEXT = "A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions.\n"

    with open(image_path, 'rb') as f:
        img_str = base64.b64encode(f.read()).decode('utf-8')
        prompt = CONTEXT + f''### Human: What is unusual about this image: \n<img src="data:image/jpeg;base64,{img_str}">### Assistant: '

        data = {
            "mode": "instruct", # chat, instruct
            "character": "Example",
#           "instruction_template": "LLaVA-v1",
            "messages": [{"role": "system", "content": prompt}]
        }

        response = requests.post('http://127.0.0.1:5000/v1/chat/completions', json=data)

        if response.status_code != 200:
            print(f"Request failed with status {response.status_code}. Response: {response.json()}")
            return None
        else:
            return response.json()['choices'][0]['message']['content']

image_path = "extreme_ironing.jpg"
text_captioning = process_image(image_path)
print(text_captioning)

```
</details>
